packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -0,0 +1,174 @@
+const contentful = require(`contentful`)
+const crypto = require(`crypto`)
+const _ = require(`lodash`)
+
+const typePrefix = `contentful__`
+const makeTypeName = type => `${typePrefix}${type.replace(/-/g, `_`)}`
+
+exports.sourceNodes = async (
+  { boundActionCreators, getNode, hasNodeChanged, store },
+  { spaceId, accessToken }
+) => {
+  const {
+    createNode,
+    updateSourcePluginStatus,
+    touchNode,
+  } = boundActionCreators
+  updateSourcePluginStatus({
+    plugin: `gatsby-source-contentful`,
+    status: {
+      ...store.getState().status[`gatsby-source-contentful`],
+      ready: false,
+    },
+  })
+
+  // Touch existing Contentful nodes so Gatsby doesn't garbage collect them.
+  _.values(store.getState().nodes)
+    .filter(n => n.internal.type.slice(0, 8) === typePrefix)
+    .forEach(n => touchNode(n.id))
```

Comment:
This isn't what we want while we're not using the sync API. Right now an entry could be deleted from Contentful and not be removed from the Gatsby site. Touching is only necessary if we retrieve from a remote API only the updated data so we "touch" the nodes which haven't updated but we know also weren't deleted. And oh lookie, there's even docs for this now :D https://www.gatsbyjs.org/docs/bound-action-creators/#touchNode